mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -72,22 +77,39 @@ public ExecutionStages deliver(Collection<MailboxListener> mailboxListeners, Eve
 
     private Mono<Void> doDeliver(Collection<MailboxListener> mailboxListeners, Event event) {
         return Flux.fromIterable(mailboxListeners)
-            .flatMap(mailboxListener -> Mono.fromRunnable(() -> doDeliverToListener(mailboxListener, event)))
+            .flatMap(mailboxListener -> deliveryWithRetries(event, mailboxListener))
             .then()
             .subscribeOn(Schedulers.elastic());
     }
 
+    private Mono<Object> deliveryWithRetries(Event event, MailboxListener mailboxListener) {
```

Comment:
I'm a bit uncomfortable with `Mono<Object>`, can you replace `Object` by `Void` or a precise `class`?

Reply:
sorry, Intellij was faster than my eyes :)